### PR TITLE
made the main page static and enabled scrolling in the filters component

### PR DESCRIPTION
**Pull request by mtalhabaig3**
Fixes issue #5884 

Hey @HelNershingThapa, I almost fixed the issue but there is one thing left, when we the more filter component appears, the rest of the page becomes static and scrolling is enabled in the more filter component, however the height of the component overflows below the bottom of the screen. Right now i am thinking to calculate the height of the content above and set the height of the filter component 100vh - (above content height), so that it will not overflow and stick to the height available up to the end of the screen. What do you think about it?

Kindly review 🙂

### CODE REVIEW

File: frontend/src/views/project.css
Comment (HelNershingThapa) [inline frontend/src/views/project.css:1]: Could you please add this style to the `src/assets/styles/_extra.scss` file instead?

File: frontend/src/views/project.js [hunk 1]
```diff
@@ -152,14 +153,37 @@ export const MoreFilters = () => {
...
 
   return (
     <>
       <div className="absolute left-0 z-4 mt1 w-40-l w-100 h-100 bg-white h4 ph1 ph5-l">
-        <MoreFiltersForm currentUrl={currentUrl} />
+        <div
+          className="scrollable-container"
+          style={{
+            height: '100%',
+            overflow: 'auto',
```
Comment (HelNershingThapa) [inline frontend/src/views/project.js:179]: We use utility classes from [tachyons](https://tachyons.io/), so I recommend you use corresponding utility classes from tachyons if available. These two could be just replaced with `h-100 overflow-hidden`.

File: frontend/src/views/project.js [hunk 2]
```diff
@@ -152,14 +153,37 @@ export const MoreFilters = () => {
...
+            scrollbarWidth: 'none',
+            msOverflowStyle: 'none',
```
Comment (HelNershingThapa) [inline frontend/src/views/project.js:181]: It seems like these two lines may not serve any purpose. Could you please confirm if they have any functional significance?

**Review by HelNershingThapa — COMMENTED** [review]
Your proposed solution sounds good to me and is reasonable!  Please ensure that the filter component doesn't overflow and remains scrollable within the available height up to the end of the screen.

File: frontend/src/views/project.js (hunk 1 above)
Comment (mtalhabaig3) [reply frontend/src/views/project.js:179]: Yes exactly! I will fix it.

File: frontend/src/views/project.js (hunk 2 above)
Comment (mtalhabaig3) [reply frontend/src/views/project.js:181]: Yes you are right, i will remove them.

**Comment by mtalhabaig3** [comment]
Yes, i am on it.

File: frontend/src/views/project.css
Comment (mtalhabaig3) [reply frontend/src/views/project.css:1]: Yes sure!

**Comment by mtalhabaig3** [comment]
@HelNershingThapa I have successfully solved the issue. I have set the height of the more filters menu by subtracting the inner window height by the height of the content above it. I have also added the functionality so that the height of the menu changes when we resize the window vertically. Kindly review and let me know if you want some thing changed! 🙂

**Comment by mtalhabaig3** [comment]
I have removed the duplicate call to the component and a console log.

File: frontend/src/views/project.js [hunk 3]
```diff
@@ -151,15 +151,51 @@ export const ProjectsPageIndex = (props) => {
...
+    };
+
+    handleResize();
+
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  });
```
Comment (HelNershingThapa) [inline frontend/src/views/project.js:184]: Would it be better to include a dependency array in this `useEffect` hook?

File: frontend/src/assets/styles/_extra.scss [hunk 4]
```diff
@@ -616,3 +616,7 @@ a[href="https://www.mapbox.com/map-feedback/"]
 .link:focus {
   outline: revert;
 }
+
+.scrollable-container::-webkit-scrollbar {
+  width: 0;
+}
```
Comment (HelNershingThapa) [inline frontend/src/assets/styles/_extra.scss:622]: This effectively hides the scrollbar - users should be able to identify and interact with scrollable content easily.

missing-scrollbar

**Review by HelNershingThapa — CHANGES_REQUESTED** [review]

File: frontend/src/views/project.js (hunk 3 above)
Comment (mtalhabaig3) [reply frontend/src/views/project.js:184]: Yes you are right! that will keep the effect from re rendering on every re render of the component

File: frontend/src/assets/styles/_extra.scss (hunk 4 above)
Comment (mtalhabaig3) [reply frontend/src/assets/styles/_extra.scss:622]: You mean i should remove this and keep the scroll bar visible?

Comment (HelNershingThapa) [reply frontend/src/assets/styles/_extra.scss:622]: The scroll bar should be visible when the content overflows the container.

Comment (mtalhabaig3) [reply frontend/src/assets/styles/_extra.scss:622]: Sure! I will do that.

**Comment by mtalhabaig3** [comment]
@HelNershingThapa done.

**Review by HelNershingThapa — CHANGES_REQUESTED** [review]
This is great. 

I noticed there is a horizontal scrollbar appearing in the user interface. Could you please remove the scrollbar?

scrollbar

Also, there is a failing test case due to the addition of `useEffect` code. Could you please address those failing tests as well? Thank you!

**Comment by mtalhabaig3** [comment]
@HelNershingThapa yes sure! I will finish up these things.

**Comment by HelNershingThapa** [comment]
hi @mtalhabaig3 I hope you've been well. It's been a little while since the last update on this PR. Could you please provide an update on its current status? I'm particularly interested in the progress regarding the changes you mentioned in your previous comment, such as addressing the failing tests and ensuring the improvements for the dropdown behavior.

**Comment by HelNershingThapa** [comment]
If the failing test case is causing a roadblock, you might consider adding the following line to the test:
```js
jest.spyOn(document, 'getElementById').mockReturnValue({ offsetHeight: 100 });
```

This mock should suffice for now, as it doesn't seem to interfere with other aspects of the test case. Although I'm not a Jest testing expert, this approach might help move things forward without significant hindrance.

Your efforts to resolve these issues are highly valued, and I look forward to seeing this pull request's continued progress.

**Comment by mtalhabaig3** [comment]
> hi @mtalhabaig3 I hope you've been well. It's been a little while since the last update on this PR. Could you please provide an update on its current status? I'm particularly interested in the progress regarding the changes you mentioned in your previous comment, such as addressing the failing tests and ensuring the improvements for the dropdown behavior.

**Comment by mtalhabaig3** [comment]
@HelNershingThapa Sure, Everything is fine except the failing tests. I will do as you said and let you know if it worked.

**Comment by HelNershingThapa** [comment]
hi @mtalhabaig3 I did notice a few commented out code that could benefit from some cleanup. Once those are done, I'll gladly dive into the review and merge the code.

**Comment by mtalhabaig3** [comment]
Sure @HelNershingThapa I am on it!

**Comment by mtalhabaig3** [comment]
Done!

**Comment by royallsilwallz** [comment]
Continued with #6325

**Pull request closed without merging**